presto-native-execution/src/test/java/com/facebook/presto/spark/AbstractTestPrestoSparkQueries.java

```diff
@@ -0,0 +1,92 @@
+import com.facebook.presto.testing.ExpectedQueryRunner;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.google.common.collect.ImmutableMap;
+
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.SystemSessionProperties.NATIVE_EXECUTION_ENABLED;
+import static com.facebook.presto.SystemSessionProperties.NATIVE_EXECUTION_EXECUTABLE_PATH;
+import static com.facebook.presto.SystemSessionProperties.TABLE_WRITER_MERGE_OPERATOR_ENABLED;
+import static com.facebook.presto.hive.HiveSessionProperties.COLLECT_COLUMN_STATISTICS_ON_WRITE;
+import static com.facebook.presto.spark.PrestoSparkSessionProperties.SPARK_INITIAL_PARTITION_COUNT;
+import static com.facebook.presto.spark.PrestoSparkSessionProperties.SPARK_PARTITION_COUNT_AUTO_TUNE_ENABLED;
+import static java.util.Objects.requireNonNull;
+
+public class AbstractTestPrestoSparkQueries
+        extends AbstractTestQueryFramework
+{
+    private static final String SPARK_SHUFFLE_MANAGER = "spark.shuffle.manager";
+    private static final String FALLBACK_SPARK_SHUFFLE_MANAGER = "spark.fallback.shuffle.manager";
+
+    protected Session getNativeSession()
+    {
+        Session.SessionBuilder sessionBuilder = Session.builder(getSession())
+                .setSystemProperty(NATIVE_EXECUTION_ENABLED, "true")
+                .setSystemProperty(TABLE_WRITER_MERGE_OPERATOR_ENABLED, "false")
+                .setSystemProperty(SPARK_INITIAL_PARTITION_COUNT, "1")
+                .setSystemProperty(SPARK_PARTITION_COUNT_AUTO_TUNE_ENABLED, "false")
+                .setCatalogSessionProperty("hive", COLLECT_COLUMN_STATISTICS_ON_WRITE, "false");
+
+        if (System.getProperty("NATIVE_PORT") == null) {
+            sessionBuilder.setSystemProperty(NATIVE_EXECUTION_EXECUTABLE_PATH, requireNonNull(System.getProperty("PRESTO_SERVER"), "Native worker binary path is missing. Add -DPRESTO_SERVER=/path/to/native/process/bin to your JVM arguments."));
+        }
+        return sessionBuilder.build();
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+    {
+        // prevent to use the default Prestissimo config files since the Presto-Spark will generate the configs on-the-fly.
+        return PrestoSparkNativeQueryRunner.createPrestoSparkNativeQueryRunner(ImmutableMap.of("catalog.config-dir", "/"));
+    }
+
+    @Override
+    protected ExpectedQueryRunner createExpectedQueryRunner()
+            throws Exception
+    {
+        String dataDirectory = System.getProperty("DATA_DIR");
+        return HiveExternalWorkerQueryRunner.createJavaQueryRunner(Optional.ofNullable(dataDirectory).map(Paths::get));
+    }
+
+    protected void assertQuery(String sql)
+    {
+        // Reset the spark context to register the native execution shuffle manager. We want to let the query runner use the default spark shuffle
```

Comment:
Do we still need this? Can we use vanilla Presto Java query runner to create test tables?

Reply:
IIUC, we can only have two runners in one unit test (actual vs expected), is it feasible to run 3? -  vanilla presto runner to create table,  prestoSpark to run native, hive runner to run expected result?

Reply:
I think it is sufficient to have 2. See presto-native-execution/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java where we create test tables using the expected query runner.

Reply:
Hmm, after second look, I believe we already use the expected runner to create the test tables  at L68. At the line you're commenting, we just get the actual runner and run the test without creating the test table. Do I miss anything?